src/components/ui/BlogItem.jsx

```diff
@@ -1,11 +1,87 @@
-const Blog = ({ data: blog }) => {
+import { format } from 'date-fns';
+import lt from 'date-fns/locale/lt';
+import Image from 'next/image';
+import Link from 'next/link';
+import { shuffleArray } from './ShuffleArray';
+
+const BlogItem = ({ blog, styleLatest, index }) => {
+  const latestStyle = styleLatest && index !== 0;
+  const firstFromRecent = styleLatest && index === 0;
+  const categoryColors = [
+    'text-blue-600 bg-blue-100',
+    'text-red-600 bg-red-100 ',
+    'text-green-600 bg-green-100',
+    'text-yellow-600 bg-yellow-100',
+    'text-purple-600 bg-purple-100',
+    'text-pink-600 bg-pink-100',
+    'text-indigo-600 bg-indigo-100',
+    'text-teal-600 bg-teal-100',
+    'text-orange-600 bg-orange-100',
+    'text-gray-600 bg-gray-100',
+  ];
+  const shuffledCategoryColors = shuffleArray(categoryColors);
   return (
-    <>
-      <div>
-        <h2>{blog.title}</h2>
-        <p>{blog.content.text}</p>
+    <Link
+      href={`/blogs/#`}
+      className={`no-underline ${
+        latestStyle
+          ? 'flex flex-col md:flex-row items-center md:items-start'
+          : 'grid content-between justify-items-center md:justify-items-start'
+      } gap-6 `}
+    >
+      <Image
+        className='md:hidden m-0'
+        src={blog.photo.url}
+        alt={blog.photo.fileName}
+        width={684}
+        height={456}
+        sizes='sizes="(min-width: 1480px) 419px, (min-width: 1040px) 28.57vw, (min-width: 780px) calc(50vw - 52px), calc(100vw - 64px'
+      />
+      <Image
+        className='hidden md:block m-0'
+        src={blog.photo.url}
+        alt={blog.photo.fileName}
+        width={latestStyle ? 300 : firstFromRecent ? 1000 : 430}
+        height={latestStyle ? 210 : 230}
+        style={{
+          height: latestStyle ? '210px' : '230px',
+          maxWidth: latestStyle ? '300px' : '',
+          objectFit: 'cover',
+        }}
+        sizes='sizes="(min-width: 1480px) 419px, (min-width: 1040px) 28.57vw, (min-width: 780px) calc(50vw - 52px), calc(100vw - 64px'
+      />
+
+      <div className={`${latestStyle && 'w-full md:w-1/2'} w-full`}>
+        <div className='flex gap-x-1 items-center text-purple-700 text-xs'>
+          <span>{blog.author}</span>
+          <span>•</span>
+          <span className=' whitespace-nowrap'>
+            {format(new Date(blog.createdDate), ' d MMM  Y', { locale: lt })}
```

Comment:
extract this to a function in date utils so it can be reused in other places

Reply:
I fixed it